src/layer/S1GRDEOCloudLayer.ts

```diff
@@ -0,0 +1,105 @@
+  protected resolution: Resolution | null = null;
+
+  public constructor(
+    instanceId: string | null,
+    layerId: string | null = null,
+    evalscript: string | null = null,
+    evalscriptUrl: string | null = null,
+    title: string | null = null,
+    description: string | null = null,
+    acquisitionMode: AcquisitionMode | null = null,
+    polarization: Polarization | null = null,
+  ) {
+    super(instanceId, layerId, evalscript, evalscriptUrl, title, description);
+    // it is not possible to determine these parameters by querying the service, because there
+    // is no endpoint which would return them:
+    if ((evalscript || evalscriptUrl) && (!acquisitionMode || !polarization)) {
+      throw new Error('Parameters acquisitionMode and polarization are mandatory when using evalscript');
+    }
+    this.acquisitionMode = acquisitionMode;
+    this.polarization = polarization;
+  }
+
+  public static makeLayer(
+    layerInfo: any,
+    instanceId: string,
+    layerId: string,
+    evalscript: string | null,
+    evalscriptUrl: string | null,
+    title: string | null,
+    description: string | null,
+  ): S1GRDEOCloudLayer {
+    let acquisitionMode = null;
+    let polarization = null;
+    switch (layerInfo.settings.datasourceName) {
+      case 'S1':
+        acquisitionMode = AcquisitionMode.IW;
+        polarization = Polarization.DV; // SV is not available on EO Cloud
+        break;
+      case 'S1_EW':
+        acquisitionMode = AcquisitionMode.EW;
+        polarization = Polarization.DH;
+        break;
+      case 'S1_EW_SH':
+        acquisitionMode = AcquisitionMode.EW;
+        polarization = Polarization.SH;
+        break;
+      default:
+        throw new Error(`Unknown datasourceName (${layerInfo.settings.datasourceName})`);
+    }
+    return new S1GRDEOCloudLayer(
+      instanceId,
+      layerId,
+      evalscript,
+      evalscriptUrl,
+      title,
+      description,
+      acquisitionMode,
+      polarization,
+    );
+  }
+
+  protected getEvalsource(): string {
+    // ignore this.dataset.shWmsEvalsource and return the string based on acquisitionMode:
+    if (this.acquisitionMode === AcquisitionMode.IW && this.polarization === Polarization.DV) {
+      // note that on EO Cloud, for IW acquisition mode only DV is available (SV is not available)
+      return 'S1';
+    }
+    if (this.acquisitionMode === AcquisitionMode.EW && this.polarization === Polarization.DH) {
+      return 'S1_EW';
+    }
+    if (this.acquisitionMode === AcquisitionMode.EW && this.polarization === Polarization.SH) {
+      return 'S1_EW_SH';
+    }
+    throw new Error(
+      `This combination of acquisition mode and polarization (${this.acquisitionMode} / ${
+        this.polarization
+      }) is not available on EO Cloud`,
+    );
+  }
+
+  protected getFindTilesAdditionalParameters(): Record<string, string> {
+    return {
+      productType: 'GRD',
```

Comment:
What is the `productType` intended for? Because I can't find it anywhere else in this MR and in other files there is only `type` in `S1GRDAWSEULayer` and `S3SLSTRLayer`.

Reply:
Function `getFindTilesAdditionalParameters()` returns the parameters that are then sent to the backend service and `productType: GRD` is one such parameter...

Reply:
Ah true. I just overlooked that the `productType` is a parameter for Sentinel-1 on EOCloud.